rosbag2_compression/include/rosbag2_compression/sequential_compression_reader.hpp

```diff
@@ -0,0 +1,153 @@
+    std::shared_ptr<rosbag2_cpp::SerializationFormatConverterFactoryInterface> converter_factory =
+    std::make_shared<rosbag2_cpp::SerializationFormatConverterFactory>(),
+    std::unique_ptr<rosbag2_storage::MetadataIo> metadata_io =
+    std::make_unique<rosbag2_storage::MetadataIo>());
+
+  virtual ~SequentialCompressionReader();
+
+  void open(
+    const rosbag2_cpp::StorageOptions & storage_options,
+    const rosbag2_cpp::ConverterOptions & converter_options) override;
+
+  void reset() override;
+
+  bool has_next() override;
+
+  std::shared_ptr<rosbag2_storage::SerializedBagMessage> read_next() override;
+
+  std::vector<rosbag2_storage::TopicMetadata> get_all_topics_and_types() override;
+
+  /**
+   * Ask whether there is another database file to read from the list of relative
+   * file paths.
+   *
+   * \return true if there are still files to read in the list
+   */
+  virtual bool has_next_file() const;
+
+  /**
+   * Return the relative file path pointed to by the current file iterator.
+   */
+  virtual std::string get_current_file() const;
+
+  /**
+   * Return the URI of the current file (i.e. no extensions).
+   */
+  virtual std::string get_current_uri() const;
+
+protected:
+  /**
+   * Increment the current file iterator to point to the next file in the list of relative file
+   * paths.
+   *
+   * Expected usage:
+   * if (has_next_file()) load_next_file();
+   */
+  virtual void load_next_file();
+
+  /**
+   * Initializes the decompressor if a compression mode is specified in the metadata.
+   *
+   * \throws std::invalid_argument If compression format doesn't exist.
+   */
+  virtual void setup_compression();
+
+  /**
+   * Set the file path currently pointed to by the iterator.
+   *
+   * \param file Relative path to the file.
+   */
+  virtual void set_current_file(const std::string & file);
+
+private:
+  /**
+   * Checks if all topics in the bagfile have the same RMW serialization format.
+   * Currently a bag file can only be played if all topics have the same serialization format.
+   *
+   * \param topics Vector of TopicInformation with metadata.
+   * \throws runtime_error if any topic has a different serialization format from the rest.
+   */
+  virtual void check_topics_serialization_formats(
+    const std::vector<rosbag2_storage::TopicInformation> & topics);
+
+  /**
+   * Checks if the serialization format of the converter factory is the same as that of the storage
+   * factory.
+   * If not, changes the serialization format of the converter factory to use the serialization
+   * format of the storage factory.
+   *
+   * \param converter_serialization_format
+   * \param storage_serialization_format
+   */
+  virtual void check_converter_serialization_format(
+    const std::string & converter_serialization_format,
+    const std::string & storage_serialization_format);
+
+  std::unique_ptr<rosbag2_storage::StorageFactoryInterface> storage_factory_{};
```

Comment:
Nit: some `std::unique_ptr`s are default constructed (initialized with `{}`) and some are initialized to `nullptr`. Actually, both are unnecessary as C++ classes like smart pointers are well-behaved, unlike POD types like raw pointers.

Reply:
Moving in favor of `{}` so all fields in `SequentialCompressionReader` are initialized similarly.

Reply:
My suggestion is to make them all consistent, with either no initializer or with empty brace initializer (`{}`).